File: 백준/Silver/10866. 덱/덱.py
import sys
from collections import deque
n=int(sys.stdin.readline())
dq= deque()
for i in range(n):
    com=sys.stdin.readline().split()
    if(com[0]=="push_front"):
        dq.appendleft(int(com[1]))
    elif(com[0]=="push_back"):
        dq.append(int(com[1]))
    elif(com[0]=="pop_front"):
        if(dq):
            print(dq.popleft())
        else:
            print(-1)
    elif(com[0]=="pop_back"):
        if(dq):
            print(dq.pop())
        else:
            print(-1)
    elif(com[0]=="size"):
        print(len(dq))
    elif(com[0]=="empty"):
        if(dq):
            print(0)
        else:
            print(1)
    elif(com[0]=="front"):
        if(dq):
            print(dq[0])
        else:
            print(-1)
    elif(com[0]=="back"):
            if(dq):
                print(dq[-1])
            else:
                print(-1)

File: 백준/Silver/10866. 덱/덱.cc
#include <stdio.h>
#include <iostream>
#include <string>
#include <deque>
#include <utility>
int main(){
    std::cin.tie(0);
    std::ios::sync_with_stdio(0);
    std::deque <int> dq{};
    int n{}, num{};
    std::cin >> n;
    std::string com{};
    for(int i=0; i<n; i++){
        std::cin >> com;
        if(com=="push_front"){
            std::cin >> num;
            dq.push_front(num);
        }
        else if(com=="push_back"){
            std::cin >> num;
            dq.push_back(num);
        }
        else if(com=="pop_front"){
            if(!dq.empty()){
                std::cout << dq.front() <<"\n";
                dq.pop_front();
            }
            else{
                std::cout << -1<<"\n";
            }
        }
        else if(com=="pop_back"){
            if(!dq.empty()){
                std::cout << dq.back() <<"\n";
                dq.pop_back();
            }
            else{
                std::cout << -1<<"\n";
            }
        }
        else if(com=="size"){
            std::cout << dq.size() <<"\n";
        }
        else if(com=="empty"){
            std::cout << dq.empty() <<"\n";
        }
        else if(com=="front"){
            if(!dq.empty()){
                std::cout << dq.front() <<"\n";
            }
            else{
                std::cout << -1<<"\n";
            }
        }
        else if(com=="back"){
            if(!dq.empty()){
                std::cout << dq.back() <<"\n";
            }
            else{
                std::cout << -1<<"\n";
            }
        }
    }
}